.github/workflows/compile-rocm.yaml

```
@@ -0,0 +1,30 @@
+name: GitHub Action CI
+
+on: [pull_request]
+
+env:
+  ROCM_VER: 5-4
+jobs:
+  compile-rocm:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends wget lsb-core software-properties-common gpg curl
+    - name: Install extra dependencies
+      run: |
+        curl -fsSL https://repo.radeon.com/rocm/rocm.gpg.key | sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/rocm-keyring.gpg
+        echo 'deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/rocm-keyring.gpg]  https://repo.radeon.com/rocm/apt/debian focal main' | sudo tee /etc/apt/sources.list.d/rocm.list
+        sudo apt-get update
+        sudo apt-get install -y rocm-hip-sdk
+    - uses: actions/checkout@v2
+      with:
+            submodules: true
+    - name: Build Open MPI
+      run: |
+        git submodule update --init --recursive
```

You should be able to use `submodules: recursive` and then not need the `git submodule update...` line in your `run` stanza.

```suggestion
            submodules: recursive
    - name: Build Open MPI
      run: |
 ```

---

> note: the only reason that I do not simply sign off and commit your change suggestion using the button on the github webpage is because of the weird sign-off line that it wants to add

No worries; I just used that "suggested change" syntax because it seemed a convenient way to show exactly what I was suggesting.